### ISSUE TITLE
Different settings for default viewport start position

### ISSUE THREAD

**Issue by DarkDefender** [eXeC64/imv#176]
I'm using imv to read pages from comics or simply extracts from news papers. However, it is quite annoying to have to scroll up to the top of the page on each image change, so I made this current hack:

```
diff --git a/src/viewport.c b/src/viewport.c
index 3d3251b..99d3784 100644
--- a/src/viewport.c
+++ b/src/viewport.c
@@ -174,7 +174,8 @@ void imv_viewport_center(struct imv_viewport *view, const struct imv_image *imag
   const int image_height = imv_image_height(image);

   view->x = (ww - image_width * view->scale) / 2;
-  view->y = (wh - image_height * view->scale) / 2;
+  //view->y = (wh - image_height * view->scale) / 2;
+  view->y = 0;

   view->locked = 1;
   view->redraw = 1;
```

But a better solution to this would perhaps to have options for the default position. So perhaps something like:
```
_____
1|2|3
-----
4|5|6
-----
7|8|9
-----

1: top_left
2: top_center
3: top_right
4: center_left
5: center
6: center_right
7: bottom_left
8: bottom_center
9: bottom_right
```

Or I guess perhaps just exposing two percentage variables would be better (?)

So:
```
X=0%,Y=0% == top_left

X=50%,Y=0% == top_center

etc
```

**Comment by eXeC64** [comment]
That's an interesting use case, I think something like this makes sense to do. Would an additional scaling mode also be of use? A new one called `fill` or `crop` instead of `full`, which zooms in until the image fills the screen in both axes, even if it doesn't all fit on screen.

The idea being for a single comic page, you could start with the top half filling your screen, and then just scroll down with the mouse wheel.

**Comment by DarkDefender** [comment]
Yes, more scaling modes would be awesome to have. Currently it is a bit painful to read really high res stuff as the current scaling modes doesn't really work that well for reading.

:+1:

**Comment by rien333** [comment]
A way to navigate image positions (e.g. top/bottom) would be super handy when dealing with charts and informative images! Really something I miss in `imv` when compared to other minimal/vim inspired apps like zathura and qutebrowser.

**Comment by eXeC64** [comment]
@rien333 How do you mean? Something like `pan 0 top` to stay in the current x position, but line up with the top of the image, or `pan center bottom` to go to the centre bottom edge?

This would be to go with a `initial_pan = <left|center|right> <top|center|bottom>` config option of some sort. Names to be decided.

**Comment by rien333** [comment]
All of that sounds great, but I would slight prefer to have an option to keep the current x/y position (so something like `pan 0 top`, or slightly simpler just  `pan <left|right|bottom|top>` and always keep the current x/y position).

**Comment by eXeC64** [comment]
Since the work is being done for me, I suppose this can sneak into v4.0

**Comment by eXeC64** [comment]
Resolved by #183 and #186
